doc/contributing/result-object-and-error-handling.md

```diff
@@ -0,0 +1,77 @@
+# Result object and Error Handling
+
+
+## Result object
+
+The Result class is similar to Option type/object in Scala/Rust.
+It wraps up some `value` and `error` - so that a function can return both simulateously.
+
+This is similar to the https://www.npmjs.com/package/neverthrow package
+
+### Simple usage
+
+```ts
+function someFn(someArg): Result<string | undefined> {
+  if (someArg) {
+    return new Result('success')
+  }
+  return new Result(undefined, TerriaError.from("SOME ERROR"))
+}
+```
+
+#### This can now be used in a few different ways
+
+##### Ignore error and just return value
```

Comment:
For each of these examples, you should add a comment showing what the value of `value` is.